plugins/mongo_db_plugin/mongo_db_plugin.cpp

```diff
@@ -779,8 +779,8 @@ void mongo_db_plugin_impl::_process_accepted_transaction( const chain::transacti
    }
 
    string signing_keys_json;
-   if( t->signing_keys.valid() ) {
-      signing_keys_json = fc::json::to_string( t->signing_keys->second );
+   if( t->signing_keys_future.valid() ) {
```

Comment:
This logic would be simplified if `recover_keys` synchronously recovered the keys with no time limit.